[agreatdayy] ip

agreatdayy opened the pull request:
# ChrisPBacon
> “Oink oink!” – Chris P Bacon

Oink Oink! ChrisPBacon frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let ChrisP manage your tasks for you 😉

And it is **FREE**! Oink!!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

-----------------------------------------------------------------------------------------------------
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

agreatdayy committed:
level 0

agreatdayy committed:
level 1 - echo

agreatdayy committed:
level 2 - add, list

agreatdayy committed:
Level-3: mark as done

agreatdayy committed:
Level 4: ToDos, Events, Deadlines

agreatdayy committed:
A-TextUiTesting

agreatdayy committed:
corrected some typos

agreatdayy committed:
removed blank spaces in EXPECTED.TXT

agreatdayy committed:
Level 5: Handle errors

agreatdayy committed:
updated test

agreatdayy committed:
Level-6:Delete

agreatdayy committed:
updated expected results

agreatdayy committed:
Add InvalidTaskNameException

agreatdayy committed:
Level-7: save and load

agreatdayy committed:
Merge pull request #1 from agreatdayy/branch-Level-7

Merge branch-level-7

agreatdayy committed:
Level-7: save and load

agreatdayy committed:
Merge branch 'master' of https://github.com/agreatdayy/ip

agreatdayy committed:
Level-8:date

agreatdayy committed:
Merge pull request #2 from agreatdayy/branch-Level-8

Level-8: Date for deadline tasks

agreatdayy committed:
MoreOOP

agreatdayy committed:
Merge pull request #3 from agreatdayy/A-MoreOOP

Add More OOP

Ma-Yirui commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,74 @@
+                default:
+                    // if user entered input that cannot be recognised.
+                    throw new DukeException("Ooink oink! I'm sorry, I don't understand.\n"
+                            + "Type 'help' for command info!\n");
```
Perhaps adding break statement or //Fallthrough comments so that it is in line with the java coding standard?
Though it does not affect the functionality of the code.

Ma-Yirui commented on src/main/java/Gui.java:
```diff
@@ -0,0 +1,204 @@
+ *  messages in the console to the user.
+ */
+public class Gui extends Duke {
+    private final String line = "_____________________________________________________";
```
I suppose "line" is a constant. 
You can consider making line static final and name it in all uppercase letters.

Ma-Yirui commented on src/main/java/SaveAndLoad.java:
```diff
@@ -0,0 +1,48 @@
+                boolean isDone = str.charAt(1) == 'x';
+                String task = str.substring(7);
+                switch (type) {
+                    case 'T':
+                        taskList.add(new Todo(task, isDone));
+                        break;
+                    case 'D':
+                        taskList.add(new Deadline(task, isDone));
+                        break;
+                    case 'E':
+                        taskList.add(new Event(task, isDone));
+                        break;
```
There should be no indentation for case clauses according to the coding standard.

Ma-Yirui reviewed (commented):
Great job overall. Most codes are in line with the java coding standard and are easy to read.
Just several minor inconsistencies with the standard need to be fixed.
Keep up the good work!

agreatdayy committed:
Add Task and Duke packages

agreatdayy committed:
Merge pull request #4 from agreatdayy/A-Packages

Add Task and Duke packages

agreatdayy committed:
Merge pull request #5 from agreatdayy/add-gradle-support

Add gradle support

agreatdayy committed:
Set up gradle

agreatdayy committed:
Merge pull request #6 from agreatdayy/A-Gradle

Set up gradle

agreatdayy committed:
Rename packages

agreatdayy committed:
bug fix: moved all packages into seedu folder

tamagochuuu commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -0,0 +1,204 @@
+ *  messages in the console to the user.
+ */
+public class Ui extends TaskList {
+    private final String line = "_____________________________________________________";
```
According to the coding style for this module, 'line' is a constant and hence should be named LINE instead!

tamagochuuu commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,70 @@
+        super(tasks);
+    }
+
+    public void parse() {
```
This method is public, and hence should be commented with javadocs, according tot he module's coding conventions

tamagochuuu commented on src/main/java/seedu/duke/Storage.java:
```diff
@@ -0,0 +1,60 @@
+            char type = str.charAt(4);
+            boolean isDone = str.charAt(1) == 'x';
+            String task = str.substring(7);
+            switch (type) {
+                case 'T':
+                    tasks.add(new Todo(task, isDone));
+                    break;
+                case 'D':
+                    tasks.add(new Deadline(task, isDone));
+                    break;
+                case 'E':
+                    tasks.add(new Event(task, isDone));
+                    break;
```
According to the course's coding conventions, çase'statements should not be indented

tamagochuuu reviewed (commented):
Overall very nice code, and chrisP Bacon is oink oink

agreatdayy committed:
Add JUnit tests

agreatdayy committed:
Merge pull request #7 from agreatdayy/A-JUnit

Add JUnit tests

agreatdayy committed:
Rename duke to chrisPBacon

agreatdayy committed:
ChrisPBacon: add javadoc

agreatdayy committed:
Storage: add javadoc

agreatdayy committed:
Parser: add javadoc

agreatdayy committed:
Ui: add javadoc

agreatdayy committed:
ChrisPBaconException: rename from duke and add javadoc

agreatdayy committed:
InvalidTaskNameException: add javadoc

agreatdayy committed:
Deadline: add javadoc

agreatdayy committed:
Event: add javadoc

agreatdayy committed:
Task: add javadoc

agreatdayy committed:
TaskList: add javadoc

agreatdayy committed:
Todo: add javadoc

agreatdayy committed:
Merge pull request #8 from agreatdayy/branch-A-JavaDoc

Add JavaDocs

agreatdayy committed:
Rename static final variables

agreatdayy committed:
Merge pull request #10 from agreatdayy/branch-A-CodingStandard

Rename static final variables

agreatdayy committed:
Abstract methods from Ui to Parser

agreatdayy committed:
Forgot to make new branch-Level-9

agreatdayy committed:
Add Find function

agreatdayy committed:
Merge pull request #11 from agreatdayy/branch-Level-9

Add Find function

agreatdayy committed:
README: update chatbot name

agreatdayy committed:
Add files: checkstyle.xml and suppressions.xml

agreatdayy committed:
Merge pull request #12 from agreatdayy/A-CheckStyle

Add files: checkstyle.xml and suppressions.xml

agreatdayy committed:
build.gradle: Set up JavaFX

agreatdayy committed:
Add Launcher.java

agreatdayy committed:
Add new package: util

agreatdayy committed:
Set up JavaFX

agreatdayy committed:
Add JavaFX Gui

agreatdayy committed:
Add Gui images and xml files

agreatdayy committed:
Add Launcher for Gui app

agreatdayy committed:
Change all methods to return String

agreatdayy committed:
Update list

agreatdayy committed:
Improve UI

agreatdayy committed:
Improve Gui layout

agreatdayy committed:
Merge pull request #13 from agreatdayy/branch-Level-10

Branch level 10

agreatdayy committed:
Add assertions in multiple files

There might possibly be bugs in the code created by me.

Assertions are used to define assumptions about the program state.

This is so that the runtime can verify them.

agreatdayy committed:
Merge pull request #14 from agreatdayy/branch-A-Assertions

Add assertions in multiple files

agreatdayy committed:
Bug fix: storage class

Modifications on the task list are not being updated in data/list.txt.

Added storage.save() in printTodo(), printDeadline(), printEvent(),
printDelete(), printMark(), printUnmark().

This will update data/list.txt everytime the list is modified.

agreatdayy committed:
Merge pull request #15 from agreatdayy/branch-A-CodeQuality

Improve code quality

agreatdayy committed:
Set up for C-Help

agreatdayy committed:
Make popup window for help list

agreatdayy committed:
Merge pull request #16 from agreatdayy/C-Extension

Make popup window for help list

agreatdayy committed:
Add Ui.png

agreatdayy committed:
Add HelpPage.png

agreatdayy committed:
Update README.md

agreatdayy committed:
Add HelpPage.java

agreatdayy committed:
Merge branch 'master' of https://github.com/agreatdayy/ip

agreatdayy committed:
Tweak formatting in README

agreatdayy committed:
Merge pull request #17 from agreatdayy/Published-UG

Tweak formatting in README
